snap/snapcraft.yaml

```diff
@@ -0,0 +1,135 @@
+name: opensearch # you probably want to 'snapcraft register <name>'
+base: core20 # the base snap is the execution environment for this snap
+
+version: '2.2.0' # just for humans, typically '1.2+git' or '1.3.2'
+
+summary: 'OpenSearch: community-driven, Apache 2.0-licensed search and analytics suite.'
+description: |
+    OpenSearch is a community-driven, Apache 2.0-licensed open source search and 
+    analytics suite that makes it easy to ingest, search, visualize, and analyze data. 
+    Developers build with OpenSearch for use cases such as application search, 
+    log analytics, data observability, data ingestion, and more.
+
+grade: devel # must be 'stable' to release into candidate/stable channels
+
+confinement: devmode # use 'strict' once you have the right plugs and slots
```

Comment:
the alternative here is to keep devmode and test install with jailmode https://snapcraft.io/docs/install-modes#heading--jailmode

During devmode devel you can snap install snappy-debug and run snappy-debug.scanlog in parallel to see what potential adjustments you might need to make to the snap for it to function correctly when strictly confined https://snapcraft.io/docs/debug-snaps#heading--snappy-debug

Reply:
I am seeing the value of having a `--classic` confinement instead of a `strict` one. Opensearch and the jvm rely heavily on monitoring the system resources and I think it just makes sense in this case to give it free access as opposed to looking for ways to allow those one by one. What do you think?